### ISSUE TITLE
Error "Index was out of range"

### ISSUE THREAD

**Issue by Lagaffe** [Bilal-S/iis2tomcat#24]
We've discovered an issue for BonCode to connect to Tomcat when the IIS siteId is higher then the number of sites in IIS.

This is the error displayed in the browser:
```
Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index    at BonCodeIIS.WebManagement.GetVirtualDirectories(Int16 siteId)
   at BonCodeIIS.BonCodeCallHandler.GetVDirs(Int16 siteId)
   at BonCodeIIS.BonCodeCallHandler.ProcessRequest(HttpContext context)
```

No error is logged to the logs.

Our siteId was 13, when we had 10 sites running. Changing the siteId to 10 (1 till 9 were taken, the gap was between 9 and 13) fixed the problem.

**Comment by Bilal-S** [comment]
Guust,
thanks for sharing. I would think what we're running into is the second
part of error message "Must be less than the size of the collection". When
you have 10 sites you cannot use a number grater 10? Did it work when you
set site id to 14?? . I may need to think about alternate ways besides
array pointers to find the right site if the siteIds are assigned by users.

On Thu, Jun 11, 2015 at 5:32 AM, Guust Nieuwenhuis <notifications@github.com

**Comment by Bilal-S** [comment]
This has been corrected in release 1.0.24.
You should be able to use user assigned Site Ids with different values from IIS generated ones.
Thanks again.

**Comment by Lagaffe** [comment]
Thank you Bilal!
